src/util/address.rs

```diff
@@ -676,6 +676,10 @@ impl Address {
             if script.witness_version() == Some(WitnessVersion::V0) && !(script.is_v0_p2wpkh() || script.is_v0_p2wsh()) {
                 return None
             }
+
+            if script.witness_version() == Some(WitnessVersion::V1) && !script.is_v1_p2tr() {
```

Comment:
This was discussed extensively previously. I will try to dig up references. Doing this is wrong because non-32 bytes WitnessVersion::V1 address types are still available for future usage.